From 4f43dd89a1f9c21f66d6b460c0ad1b271ee40492 Mon Sep 17 00:00:00 2001
From: Lukas Nykryn <lnykryn@redhat.com>
Date: Mon, 25 May 2020 11:59:31 +0200
Subject: [PATCH] Revert "[RHEL] network: default to network-legacy even in
 presence of nm-initrd-generator"

This reverts commit 7843bc52777683f6a4ef953d8dde7171ff3b911c.

This will cause that NM will be used in initrd on systems where it is installed.

Resolves: #1839706
---
 modules.d/40network/module-setup.sh | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/modules.d/40network/module-setup.sh b/modules.d/40network/module-setup.sh
index bf252eb7..8e2a74a3 100755
--- a/modules.d/40network/module-setup.sh
+++ b/modules.d/40network/module-setup.sh
@@ -8,9 +8,9 @@ check() {
 # called by dracut
 depends() {
     echo -n "kernel-network-modules "
-    # RHEL 8.1: Default to network-legacy unless the user chose
-    # network-manager manually
-    if ! dracut_module_included "network-manager" ; then
+    if ! dracut_module_included "network-legacy" && [ -x "/usr/libexec/nm-initrd-generator" ] ; then
+        echo "network-manager"
+    else
         echo "network-legacy"
     fi
     return 0

